src/perception/lane_detection/src/lane_detection_node.cpp

```diff
@@ -0,0 +1,138 @@
+#include <chrono>
+#include <filesystem>
+#include <functional>
+#include <memory>
+#include <string>
+
+#include <cv_bridge/cv_bridge.h>
+#include <sensor_msgs/msg/compressed_image.hpp>
+#include <sensor_msgs/msg/image.hpp>
+#include "image_transport/image_transport.hpp"
+#include "rclcpp/rclcpp.hpp"
+
+#include "lane_detection_msgs/msg/lane_detection.hpp"
+#include "lane_detection_node.hpp"
+#include "lane_engine.h"
+#include "std_msgs/msg/string.hpp"
+
+#include <opencv2/opencv.hpp>
+
+#include "common_helper_cv.h"
+#include "image_processor.h"
+
+LaneDetectionNode::LaneDetectionNode() : Node("lane_detection"), count_(0) {
+  this->declare_parameter<std::string>("camera_topic", "/CAM_FRONT/image_rect_compressed");
+  this->declare_parameter<std::string>("publish_vis_topic", "/CAM_FRONT/lanes_viz");
+  this->declare_parameter<std::string>("publish_lanes_topic", "/CAM_FRONT/lanes");
+  this->declare_parameter<bool>("save_images", false);
+  this->declare_parameter<std::string>("save_dir", "/tmp");
+  this->declare_parameter<bool>("publish_source_image", false);
+
+  this->get_parameter("camera_topic", camera_topic_);
+  this->get_parameter("publish_vis_topic", publish_vis_topic_);
+  this->get_parameter("publish_lanes_topic", publish_lanes_topic_);
+  this->get_parameter("save_images", save_images_);
+  this->get_parameter("save_dir", save_dir_);
+  this->get_parameter("publish_source_image", publish_source_image_);
+
+  if (!std::filesystem::exists(save_dir_)) {
+    std::filesystem::create_directories(save_dir_);
+  }
+
+  RCLCPP_INFO(this->get_logger(), "Subscribing to camera topic: %s", camera_topic_.c_str());
+
+  subscription_ = this->create_subscription<sensor_msgs::msg::Image>(
+      camera_topic_, 10,
+      std::bind(&LaneDetectionNode::image_callback, this, std::placeholders::_1));
+
+  image_pub_ = this->create_publisher<sensor_msgs::msg::Image>(publish_vis_topic_, 10);
```

Comment:
suggestion: Have this be a `debug` field in the config. It should only be initialized and ran when that field is set to true.